### DISCUSSION TITLE
Preserving record order with spark-excel

### DISCUSSION THREAD

**Discussion by james-miles-ccy** [nightscape/spark-excel#704] [General]
I am wondering if there is a way of preserving the order of the records when reading an excel file with spark-excel. There is (as expected) row shuffling happening during the read process and I would normally get round this by firstly creating an rdd with zipwithindex(). Is there any other way of getting round this behaviour for excel files without editing source?

**Comment by nightscape** [comment]
There were some discussions about adding a row number to each row, but afair it was never implemented.
